libs/remix-ui/search/src/lib/components/results/ResultItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useEffect, useRef, useState } from 'react'
+import { SearchContext } from '../../context/context'
+import { SearchResult, SearchResultLine } from '../../types'
+import { ResultFileName } from './ResultFileName'
+import { ResultSummary } from './ResultSummary'
+
+interface ResultItemProps {
+  file: SearchResult
+}
+
+export const ResultItem = (props: ResultItemProps) => {
+  const { state, findText, disableForceReload, updateCount } = useContext(
+    SearchContext
+  )
+  const [loading, setLoading] = useState<boolean>(false)
+  const [lines, setLines] = useState<SearchResultLine[]>([])
+  const [toggleExpander, setToggleExpander] = useState<boolean>(false)
+  const reloadTimeOut = useRef(null)
+  const subscribed = useRef(true)
```

Comment:
Is there any reason why you use `useRef` hook to define a boolean value? Because i can't fine a use case for it.

Reply:
when the component unmounts the ref current is set to false preventing the promise from the useffect async call to update the state on an unmounted component.